src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -1146,6 +1146,72 @@ class TcgStatics {
   static boolean isGxPerformed(){
     bg.em().retrieveObject("gx_"+my.owner)
   }
+
+  /* General checks for attacks and abilities */
+
+  //  * assertBench
+  //    - Can be used with no arguments, or with a condition and additional text included.
+  //    - If a closure is given, it'll check for any "it" element in bench returning true.
+  //    - Optional params:
+  //      + type: if set, restricts to benched Pokémon of a single specific type.
+  //      + isEX/isGX/isV: Can be expanded if needed. All of these unset will have the method search for any Pokémon no matter what, but if even a single one is set true it'll only filter those that are set true as well.
+  //      + opp: If true, checks for the opponent's bench instead of "my" bench.
+  //      + repText: If true, instead of adding cText at the end of the assert it'll be the only thing printed.
+  static void assertBench(params=[:], Closure c, String cText) {
+    def checkedBench = params.opp ? opp.bench : my.bench
+
+    def isPokeCnt = 0
+    [params.isEX, params.isGX, params.isV].each{isPokeCnt += 1}
+
+    def benchFilter = (
+      if (c == null) { true } else {
+        (
+          if (params.type) {it.types.contains(params.type)} else {true}
+        ) && (
+          if (!isPokeCnt) {true} else {
+            (params.isEX && it.pokemonEX) ||
+            (params.isGX && it.pokemonGX) ||
+            (params.isV && it.pokemonV)
+          }
+        ) && (
+          c(it)
+        )
+      }
+    )
+
+    def pokeString = "Pokémon"
+    if (isPokeCnt) (
+      pokeString = ""
+      i = 1
+      [(params.isEX, "Pokémon-EX"), (params.isGX, "Pokémon-GX"), (params.isV, "Pokémon V")].each{
+        if (it[0]) {
+          pokeString += it[1] + (if (i == isPokeCnt) "" else if (i == isPokeCnt-1) " or " else ", ")
+          i += 1
+        }
+      }
+    )
+
+    def failMessage = if (params.repText) { cText } else {
+      "You" + (params.opp ? "r opponent does" : " do") + "n't have any " + (
+        param.type ? params.type.getShortNotation() + " " : ""
+      ) + "Benched $pokeString" + () + (
+        if (c != null) {
+          " that " + (cText ? cText : "follow the stated condition(s)")
+        }
```

Comment:
Please type full strings without splitting them up for english grammar like this. It will be more readable/maintainable. 

Also, please use groovy Gstring feature - use variable interpolation with `$` instead of concetenation with `+`. This will also produce more readable/maintainable code.

Thanks

Reply:
> Please type full strings without splitting them up for english grammar like this. It will be more readable/maintainable.

Fair point, might be longer but also simpler to follow.

> Also, please use groovy Gstring feature - use variable interpolation with $ instead of concetenation with +. This will also produce more readable/maintainable code.

Initially went with it, but for some reason (I think a recurring error while working on some card implementations) I jumped back into concatenation. Will update it in a bit.